pandas/core/frame.py

```diff
@@ -2990,15 +2990,24 @@ def duplicated(self, subset=None, keep='first'):
         from pandas.core.groupby import get_group_index
         from pandas.core.algorithms import factorize
         from pandas.hashtable import duplicated_int64, _SIZE_HINT_LIMIT
+        from pandas.compat import long
 
         def f(vals):
-
-            # if we have integers we can directly index with these
+            # if we have integers we may be able to directly index with these
             if com.is_integer_dtype(vals):
-                from pandas.core.nanops import unique1d
-                labels, shape = vals, unique1d(vals)
-            else:
-                labels, shape = factorize(vals, size_hint=min(len(self), _SIZE_HINT_LIMIT))
+                vals = vals.astype('i8', copy=False)
```

Comment:
why are you doing all these `lower`, `upper` thing? in any case you need to factorize integers because you may not have all the integers between min and max value; 

all you need is to undo https://github.com/pydata/pandas/pull/10917 :

```diff
diff --git a/pandas/core/frame.py b/pandas/core/frame.py
index 827373c..08d2857 100644
--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -2992,13 +2992,7 @@ class DataFrame(NDFrame):
         from pandas.hashtable import duplicated_int64, _SIZE_HINT_LIMIT
 
         def f(vals):
-
-            # if we have integers we can directly index with these
-            if com.is_integer_dtype(vals):
-                from pandas.core.nanops import unique1d
-                labels, shape = vals, unique1d(vals)
-            else:
-                labels, shape = factorize(vals, size_hint=min(len(self), _SIZE_HINT_LIMIT))
+            labels, shape = factorize(vals, size_hint=min(len(self), _SIZE_HINT_LIMIT))
             return labels.astype('i8',copy=False), len(shape)
 
         if subset is None:

```